src/handlers/initiateDataRequest.ts

```diff
@@ -0,0 +1,29 @@
+import { APIGatewayProxyResult, APIGatewayProxyEvent } from 'aws-lambda'
+import { initiateS3DataCheck } from '../services/initateS3DataCheck'
+import { updateZendeskTicket } from '../services/updateZendeskTicket'
+import { validateZendeskRequest } from '../services/validateZendeskRequest'
+
+export const handler = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  console.log('received Zendesk webhook', JSON.stringify(event, null, 2))
+  const validatedZendeskRequest = validateZendeskRequest(event.body)
+  // first step
+  if (!validatedZendeskRequest.isValid) {
+    // inform Zendesk
+    updateZendeskTicket(event.body, 'invalid params')
+    return {
+      statusCode: 400,
+      body: JSON.stringify({
+        message: 'Request parameters invalid'
+      })
+    }
+  }
+  initiateS3DataCheck(validatedZendeskRequest.dataRequestParams)
```

Comment:
Just for my own understanding really, but why would you not make this function asynchronous? I often see that functions are async in Lambda code, so wondering if there was any particular reason you wouldn't do that here?

Reply:
Yep, good point! Both those things will have some network access